src/irmin-pack/dict.ml

```diff
@@ -93,9 +93,9 @@ module Make (IO : IO.S) : S = struct
     t
 
   let close t =
+    (* If called from ext, flush was already called *)
```

Comment:
I would say: it is much better if code correctness does not depend on how it is used by others; after all, we don't know how others will use the code in the future; and verifying that it is used in a particular way can be a very expensive operation requiring traversing the entire codebase. So I would prefer correctness arguments that can be made "locally" if possible. In this particular case, my expectations surrounding close are: all associated caches are expunged, the underlying file-descr is closed, etc. Now, this implementation of close doesn't do that. It only does something if open_instances becomes 0. It doesn't even cause a flush of the IO. So already this is an extremely hard function to reason about. Again, I think writing some clear documentation about "close" is supposed to do, in dict_intf.ml, will show up all sorts of problems. Aside: I really don't like the caching of IO instances in the code, for various reasons that I can maybe explain sometime.